mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -97,8 +110,9 @@ export class MetricsPlotPanel extends React.Component {
 
   getMetrics = () => {
     /* eslint-disable no-param-reassign */
-    const selectedMetricsSet = new Set(this.state.selectedMetricKeys);
```

Comment:
Most of the changes below are replacing accesses to `this.state` with accesses to URL-derived state from `this.getUrlState()`